### DISCUSSION TITLE
Default Tracking set False

### DISCUSSION THREAD

**Discussion by khalidzaibi** [ash-jc-allen/short-url#155] [Q&A]
'tracking' => [
'default_enabled' => false,
]
why it save null value into this short_url_visits table into database, as i set it false.

**Comment by ash-jc-allen** [comment]
If you've disabled the tracking, the tracking fields won't be set in the `short_url_visits` table. If you'd like to track the visits, you'll need to enable the tracking either in the config or when creating your short URLs.

I hope this helps a bit? 🙂

**Reply by lucamago** [reply]
I think he wanted to say that if I set false on tracking enabled, I'd expect that there aren't any entry written on the short_url_visits table in the database; there's a way to disabled that writing on the table completely, if the tracking is disabled? Thanks!

**Reply by ash-jc-allen** [reply]
Hey! Ah yeah, I see what you mean now. I think it might have been bad naming on my behalf when I first made the package. At the moment, there isn't a way to prevent recording the visits.

I've not really got any plans to add this feature at the moment. But if this is something enough people want, I might consider adding it in the future 😄

**Reply by lucamago** [reply]
Looking at the code I think it should be enough to add this "if" in the Resolver.php around line 60:

`if ($shortURL->track_visits) {
	$visit = $this->recordVisit($request, $shortURL);

	Event::dispatch(new ShortURLVisited($shortURL, $visit));
}`

to completely disable the tracking even on the database entry, I don't know it it's enough or there's something else :)

**Reply by onursahindur** [reply]
Thanks for a great package!

I want to steer this discussion back on track. 
@lucamago 's suggestion is looking pretty, is it possible?

Thanks.

**Reply by ash-jc-allen** [reply]
Hey @onursahindur! For the time being, I think I'm going to hold off on making a change like this. But if enough people are interested in it, I'll definitely consider adding it 😄
